packages/nextjs/components/example-ui/ContractData.tsx

```diff
@@ -33,8 +33,11 @@ export const ContractData = () => {
   useScaffoldEventSubscriber({
     contractName: "YourContract",
     eventName: "GreetingChange",
-    listener: (greetingSetter, newGreeting, premium, value) => {
-      console.log(greetingSetter, newGreeting, premium, value);
+    listener: logs => {
+      logs.map(log => {
+        const { greetingSetter, value, premium, newGreeting } = log.args;
+        console.log("📡 GreetingChange event", greetingSetter, value, premium, newGreeting);
+      });
```

Comment:
@sverps any reason why we are not doing : 

```typescript
      const { args } = logs[0];
      const { greetingSetter, value, premium, newGreeting } = args;
      console.log("📡 GreetingChange event", greetingSetter, value, premium, newGreeting);

```
It was also suggested in [wagmi migration guide](https://wagmi.sh/react/migration-guide#usecontractevent)


I saw we changed it here : 
https://github.com/scaffold-eth/scaffold-eth-2/pull/352/commits/439c4f2d50d94429e61a6d8b1e03fe58c79335a5#diff-4b13aae8a79eed1d29876f892dd1b17adeace865ad0ea5deb4baf53b00e479b1

Reply:
The listener returns an array of all logs that occurred during the current [polling interval](https://viem.sh/docs/contract/watchContractEvent.html#pollinginterval-optional). So depending on how the client is configured, this could be multiple logs. To not miss logs in any scenario, best to just loop over all logs that are returned, rather than only take the first one in the array.